File: codingsample/202008/20200822/Island_of_Knowledge_6_minesweeper.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h> 

arr_arr_integer minesweeper(arr_arr_boolean matrix) {
    arr_arr_integer output = alloc_arr_arr_integer(matrix.size);
    int i,j,k,l;
    for(i=0;i<output.size;i++){
        output.arr[i] = alloc_arr_integer(matrix.arr[0].size);
        for(j=0;j<output.arr[i].size;j++) output.arr[i].arr[j] = 0;
    }
    for(i=0;i<output.size;i++){
        for(j=0;j<output.arr[0].size;j++){
            if(matrix.arr[i].arr[j]){
                for(k=i-1;k<=i+1;k++){
                    for(l=j-1;l<=j+1;l++){
                        if(k>=0&&k<=output.size-1&&l>=0&&l<=output.arr[i].size-1){
                            if(k==i&&l==j) output.arr[k].arr[l]+=0;
                            else output.arr[k].arr[l]++;
                        }
                    }
                }
            }
        }
    }
    return output;
}

File: codingsample/202008/20200822/Island_of_Knowledge_6_minesweeper.py
def minesweeper(matrix):
    output = [[0]*len(matrix[0]) for i in range(len(matrix))]
    for i in range(len(matrix)):
        for j in range(len(matrix[0])):
            if matrix[i][j]:
                for k in range(i-1,i+2):
                    for l in range(j-1,j+2):
                        if k>=0 and k<=len(output)-1 and l>=0 and l<=len(output[0])-1:
                            if k==i and l==j: output[k][l]+=0
                            else: output[k][l]+=1
    return output
